### PR TITLE
[pkg/otlp] Bump OpenTelemetry Collector dependencies to v0.60.0

### PR DESCRIPTION
**Pull request by mx-psi**
### What does this PR do?

Bump to OpenTelemetry Collector v0.60.0. Remove any usage of deprecated methods.

### Motivation

Keep up with upstream. Be able to use the modules as dependencies on the Collector.

### Additional Notes

Our linter does not catch all deprecated functions, so I tried to check this manually very carefully, but it's possible some slipped. Please review carefully comparing with the [list of deprecations for v0.60.0](https://github.com/open-telemetry/opentelemetry-collector/blob/main/CHANGELOG.md#-breaking-changes--1).

### Describe how to test/QA your changes

One specific Go module from the OpenTelemetry Collector dependencies (pdata) will have its first release candidate soon. This means we want to thoroughly test this release to provide feedback upstream.

On the metrics side we want to use an OpenTelemetry language library (e.g. the Go library) or another OTLP-producing application to:
- [x] Send an OTLP Gauge using the Gauge instrument and test that the mapping works as expected
- [x] Send an OTLP Histogram using the Histogram instrument and test that the resulting Datadog distribution has the expected values (modulo known error tolerance)
- [x] Send an OTLP cumulative monotonic Sum using a Counter instrument and test the resulting output
- [x] Send an OTLP cumulative non-monotonic Sum using an UpDownCounter instrument and test that the mapping works as expected (as a Gauge)
- [ ] Test sending at least one message via OTLP/HTTP-JSON, OTLP/HTTP-Protobuf and OTLP/gRPC.

Mappings are documented on the [OTLP metrics types](https://docs.datadoghq.com/metrics/otlp/) public documentation page.

On the traces side 
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/otlp-v0.59.0-be24aaaa593938c7.yaml [hunk 1]
```diff
@@ -8,4 +8,4 @@
 ---
 enhancements:
   - |
-    The OTLP ingest endpoint now supports the same settings and protocol as the OpenTelemetry Collector OTLP receiver v0.59.0. In particular, this drops support for consuming OTLP/JSON v0.15.0 or below payloads.
+    The OTLP ingest endpoint now supports the same settings and protocol as the OpenTelemetry Collector OTLP receiver v0.60.0. In particular, this drops support for consuming OTLP/JSON v0.15.0 or below payloads.
```
Comment (jtappa) [inline releasenotes/notes/otlp-v0.59.0-be24aaaa593938c7.yaml:11]: I think this was meant to be a new file, not overwrite the v0.59.0 file?

**Review by jtappa — DISMISSED** [review]

File: releasenotes/notes/otlp-v0.59.0-be24aaaa593938c7.yaml (hunk 1 above)
Comment (mx-psi) [reply releasenotes/notes/otlp-v0.59.0-be24aaaa593938c7.yaml:11]: No, this was intentional, I should have explained somewhere, sorry! The PR that introduced that note (#13395) is also for 7.40. If we have two notes the changelog will say something like:

> - The OTLP ingest endpoint now supports the same settings and protocol as the OpenTelemetry Collector OTLP receiver v0.59.0
> - The OTLP ingest endpoint now supports the same settings and protocol as the OpenTelemetry Collector OTLP receiver v0.60.0

which would be confusing, so I decided the best was to edit the existing release note.

**Review by gbbr — APPROVED** [review]
All good for trace.

**Review by amenasria — APPROVED** [review]
LGTM ! 🚀

File: releasenotes/notes/otlp-v0.59.0-be24aaaa593938c7.yaml (hunk 1 above)
Comment (mx-psi) [reply releasenotes/notes/otlp-v0.59.0-be24aaaa593938c7.yaml:11]: I need to merge this before the freeze. Please have a look at the explanation, if something is wrong we can fix it in a follow up PR, thanks!

**Review by jtappa — APPROVED** [review]

**Pull request merged**